### ISSUE TITLE
facet grid axis labels are None

### ISSUE THREAD

**Issue by jhamman** [xray/xray#611]
The dim names on this plot are not showing up (e.g. `None` is not right, it should be `x` and `y`):

```Python
data = (np.random.random(size=(20, 25, 12)) + np.linspace(-3, 3, 12)) # range is ~ -3 to 4
da = xray.DataArray(data, dims=['x', 'y', 'time'], name='data')
fg = da.plot.pcolormesh(col='time', col_wrap=4)
```

![fixed](https://cloud.githubusercontent.com/assets/2443309/10322611/6821bd50-6c33-11e5-96e8-5d69a5aeca21.png)

**Comment by shoyer** [comment]
This comes up in facetted plots when you don't supply `x` and `y` labels explicitly. We should be able to infer these automatically from the data array dimensions, though.

**Comment by jhamman** [comment]
I'll tag the fix for this onto #608.  It should be simple and it will allow me to rework the logic in `_infer_xy_labels()`.

**Comment by clarkfitzg** [comment]
If the dimensions are A, B, C, and the call is `da.plot.pcolormesh(col='B')` then it seemed a little ambiguous to assign `x='A', y='C'`. So another option would be to raise an exception here.

**Comment by jhamman** [comment]
closed via #608.
